src/unity/lib/annotation/annotation_base.hpp

```diff
@@ -66,6 +66,11 @@ class EXPORT AnnotationBase : public ml_model_base {
 
   std::shared_ptr<annotation_global> get_annotation_registry();
 
+  virtual void calculateSimilarItems() = 0;
+
+  virtual annotate_spec::Data getSimilarItems(annotate_spec::Datum &item,
```

Comment:
Is there a reason that this reference is non-const?  `type&` Implies that the value will get modified by the calling function; `const type&` is how it should be done if that value will not get modified.